internal/util/vcs/git/gitUtil.go

```diff
@@ -229,12 +229,20 @@ func (g *gitUtil) OpenRepository(repoPath string) (*git.Repository, error) {
 }
 
 // GetGitToken returns the GitHub token.
-func (g *gitUtil) GetGitToken(gitServiceProvider *consts.GitServiceProvider) (string, error) {
+func (g *gitUtil) GetGitToken(gitServiceProvider *consts.GitServiceProvider, tokenFromProvider string) (string, error) {
 	if gitServiceProvider == nil || *gitServiceProvider == consts.UnknownGitServiceProvider {
 		return "", cliErrs.ErrGitServiceProviderNotSupported
 	}
 
-	gitPatToken, isSet := os.LookupEnv("TF_GIT_PAT_TOKEN")
+	var gitPatToken string
```

Comment:
Please see if this can be refactored, isSet seems redundant.
```suggestion
	gitPatToken := tokenFromProvider
	if gitPatToken == "" {
		if token, exists := os.LookupEnv("TF_GIT_PAT_TOKEN"); exists {
			gitPatToken = token
		} else {
			return "", cliErrs.ErrTfGitPatTokenNotSet
		}
	}
```

Reply:
Can you please elaborate, how isSet is redundant? Here it is being set to true if we are getting the token from the provider (it doesn't depend on the env variable  in this case) and second case is that it will look up in the env var if the git pat token is set or not and take the value accordingly.